protocols/pop3/src/main/java/org/apache/james/protocols/pop3/core/DeleCmdHandler.java

```diff
@@ -61,7 +62,12 @@ public Response onCommand(POP3Session session, Request request) {
                     StringBuilder responseBuffer = new StringBuilder(64).append("Message (").append(num).append(") does not exist.");
                     return  new POP3Response(POP3Response.ERR_RESPONSE, responseBuffer.toString());
                 }
-                List<String> deletedUidList = (List<String>) session.getAttachment(POP3Session.DELETED_UID_LIST, State.Transaction);
+                List<String> deletedUidList = session.getAttachment(POP3Session.DELETED_UID_LIST, State.Transaction)
+                    .orElseGet(() -> {
+                        ArrayList<String> uidList = new ArrayList<>();

```

Comment:
Is the mutable list on purpose ?

Reply:
yes, in many parts they fetch a list, add an element, and eventually set in back.